### PR TITLE
Specify minimum rocThrust version to resolve rocPRIM dependency issue

### PR DESCRIPTION
**Pull request by dalg24**
Propose alternate resolution for #433 
See #432

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,10 @@ add_dependencies(ArborX record_hash)
 include(CMakeDependentOption)
 cmake_dependent_option(ARBORX_ENABLE_ROCTHRUST "Enable rocThrust support" ON "Kokkos_ENABLE_HIP" OFF)
 if(Kokkos_ENABLE_HIP AND ARBORX_ENABLE_ROCTHRUST)
-  find_package(rocthrust REQUIRED CONFIG)
+  # Require at least rocThrust-2.10.5 (that comes with ROCm 3.9) because
+  # rocPRIM dependency is not set properly in exported configuration for
```
Comment (aprokop) [inline CMakeLists.txt:21]: rocPRIM -> rocPrim

**Review by aprokop — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]
Fine with me.

File: CMakeLists.txt [hunk 2]
```diff
@@ -17,7 +17,10 @@ add_dependencies(ArborX record_hash)
...
+  # earlier versions
+  find_package(rocthrust 2.10.5 REQUIRED CONFIG)
```
Comment (aprokop) [inline CMakeLists.txt:23]: Wait, does this actually work? Where is 2.10.5 coming from?

**Review by aprokop — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (aprokop) [reply CMakeLists.txt:23]: I think `rocm_setup_version` is not `rocthrust` version.

Comment (dalg24) [reply CMakeLists.txt:23]: https://github.com/ROCmSoftwarePlatform/rocThrust/releases/tag/rocm-3.9.0

Comment (aprokop) [reply CMakeLists.txt:23]: I was confused, it seems that rocthrust also carries nvidia's thrust and that's where I've been looking for version.

File: CMakeLists.txt (hunk 1 above)
Comment (dalg24) [reply CMakeLists.txt:21]: No that's not how they spell it
https://github.com/ROCmSoftwarePlatform/rocPRIM

**Comment by aprokop** [comment]
@dalg24 Please confirm that you tested with rocm-3.8 and configuration failed as expected, and passed with rocm-3.9+.

**Comment by dalg24** [comment]
> @dalg24 Please confirm that you tested with rocm-3.8 and configuration failed as expected, and passed with rocm-3.9+.

Confirmed

**Review by aprokop — APPROVED** [review]

**Pull request merged**